src/algorithm/centroid.rs

```diff
@@ -6,6 +6,8 @@ use algorithm::distance::Distance;
 
 /// Calculation of the centroid.
 pub trait Centroid<T: Float> {
+    type Centroid;
```

Comment:
i originally titled this `Result`, but that's confusing since there's `std::result::Result` which is conceptually different